tests/integration/test_agent.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for jenkins-agent-k8s-operator charm."""
+
+import logging
+
+import jenkinsapi.jenkins
+from juju.application import Application
+from juju.model import Model
+
+logger = logging.getLogger()
+
+MICROK8S_CONTROLLER = "controller"
+
+
+async def test_agent_relation(
```

Comment:
Should we also run a test that executes some simple job? Just to double check that the agent is both online and can execute jobs?

Reply:
Initially I didn't added this check to the test since i feel it's more in the scope of the jenkins charm. But it's true that testing for job execution here also make sense. I added it to the test function together with running it on both focal and jammy